mypy.ini

```diff
@@ -0,0 +1,43 @@
+[mypy]
```

Comment:
as done in https://github.com/python-poetry/poetry/pull/4750, it may make more sense to configure this in `pyproject.toml`

Reply:
I agree that's a better solution, but I dont think that change belongs in this PR. happy to rebase though, if the linked PR is merged first

Reply:
oops, i see now your pr's are stacked 👍

so happy to see mypy getting in the codebase!